lib/transaction.py

```diff
@@ -511,6 +511,32 @@ def is_schnorr_signed(self, input_idx):
                        for sig in self._inputs[input_idx].get('signatures', []))
         return False
 
+    def rpa_paycode_swap_dummy_for_destination(self, rpa_dummy_address, rpa_destination_address):
+        # This method was created for RPA - reusable payment address.  
+        # It swaps out a dummy destination address for the rpa-generated address.
+        # WARNING: This is not recommended for use outside of RPA since it could
+        # be dangerous to change the destination address of a transaction.
+ 
+        # This method expects cashaddr strings for parameters rpa_dummy_address and rpa_destination_address
+  
+        x=range(len(self._outputs))          
+        for i in x: 
```

Comment:
This is just very unidiomatic for python.  You can loop over items in a list directly.  But I guess you needed the index to "write back" to the transaction.

Can you minimally get rid of `x` and just do this?

```python
for i in range(len(self._outputs)):
```

Reply:
done in 060ca8f